admin/spec/service/restart.service.spec.js

```diff
@@ -0,0 +1,67 @@
+const sinon = require('sinon')
+const moment = require('moment')
+const pupilDataService = require('../../services/data-access/pupil.data.service')
+const schoolDataService = require('../../services/data-access/school.data.service')
+const restartService = require('../../services/restart.service')
+const pinValidator = require('../../lib/validator/pin-validator')
+const pupilMock = require('../mocks/pupil')
+const schoolMock = require('../mocks/school')
+
+/* global describe, it, expect, beforeEach, afterEach, spyOn */
+
+describe('restart.service', () => {
+  let sandbox
+
+  beforeEach(() => {
+    sandbox = sinon.sandbox.create()
+  })
+
+  afterEach(() => sandbox.restore())
+
+  describe('getPupils', () => {
+    it('it should throw an error if school is not found', async () => {
+      spyOn(schoolDataService, 'findOne').and.returnValue(null)
+      try {
+        await restartService.getPupils(schoolMock._id)
+      } catch (error) {
+        expect(error.message).toBe('School [9991001] not found')
+      }
+    })
+    it('it should return a list of pupils', async () => {
+      const pupil1 = Object.assign({}, pupilMock)
+      const pupil2 = Object.assign({}, pupilMock)
+      spyOn(schoolDataService, 'findOne').and.returnValue(schoolMock)
+      spyOn(pupilDataService, 'getSortedPupils').and.returnValue([pupil1, pupil2])
+      spyOn(restartService, 'isPupilEligible').and.returnValue(true)
+      let result
+      try {
+        result = await restartService.getPupils(schoolMock._id)
+      } catch (error) {
+        expect(error).toBeUndefined()
+      }
+      expect(result.length).toBe(2)
+    })
+  })
+
+  describe('isPupilEligible', () => {
+    it('it should return false if pupil has been restarted exactly 2 times', () => {
```

Comment:
What happens if the pupil has restarted a weirdly large number of times, like 5?

Reply:
@jon-shipley I have temporary disabled this test and removed the logic for restart counts on pupil as this is changing significantly as part of the restart submission work that is currently undergoing.

Reply:
There should never be a case where the pupil restart collection will have more than 2 restart occurances for a pupil Id based on that work.